src/components/UTXOGraph/index.tsx

```diff
@@ -0,0 +1,835 @@
+    elements: [
+      {
+        data: {
+          id: cell.generatedTxHash,
+          type: 'tx',
+        },
+      },
+      {
+        data: {
+          id: 'cell',
+          type: 'cell',
+        },
+      },
+      {
+        data: {
+          id: `${cell.generatedTxHash}-edge`,
+          source: cell.generatedTxHash,
+          target: 'cell',
+        },
+      },
+      ...(cell.consumedTxHash
+        ? [
+            {
+              data: {
+                id: cell.consumedTxHash,
+                type: 'tx',
+              },
+            },
+            {
+              data: {
+                id: `${cell.consumedTxHash}-edge`,
+                source: 'cell',
+                target: cell.consumedTxHash,
+              },
+            },
+          ]
+        : []),
+    ],
+    style: [
+      // the stylesheet for the graph
+      ...cytoscapeStyles,
+      {
+        selector: 'node[type="cell"]',
+        style: {
+          width: 16,
+          height: 16,
+          'background-color': '#00cc9b',
+        },
+      },
+    ],
+    layout: {
+      name: 'grid',
+      rows: 1,
+      cols: cell.consumedTxHash ? 3 : 2,
+    },
+  })
+  return cy
+}
+
+const TxNode = ({ txHash, modalRef }: { txHash: string; modalRef?: HTMLDivElement | null }) => {
+  const query = useQuery(['transaction', txHash], async () => {
+    const transaction = await explorerService.api.fetchTransactionByHash(txHash)
+    return transaction
+  })
+  const { t } = useTranslation()
+  const blockNumberFormat = query.data?.blockNumber !== undefined ? (+query.data.blockNumber).toLocaleString('en') : ''
+  return (
+    <div className={styles.txNodeContainer}>
+      <div className={classNames(styles.txHash, 'monospace')}>
+        {t('utxo_graph.tx_hash')}:
+        <Tooltip
+          trigger="hover"
+          placement="top"
+          title={<CopyTooltipText content={txHash} />}
+          getPopupContainer={() => modalRef ?? document.body}
+        >
+          <EllipsisMiddle text={txHash} />
+        </Tooltip>
+        <Dropdown
+          menu={{
+            items: [
+              {
+                label: `${t('block.block')}: ${blockNumberFormat}`,
+                key: 'block',
+              },
+              {
+                label: (
+                  <Link to={`/transaction/${txHash}`} target="_blank">
+                    {`${t('utxo_graph.view_tx')}`}
+                  </Link>
+                ),
+                key: 'tx',
+              },
+            ],
+            onClick(e) {
+              e.domEvent.stopPropagation()
+            },
+          }}
+          trigger={['click']}
+          getPopupContainer={() => modalRef ?? document.body}
+        >
+          <MoreAction />
+        </Dropdown>
+      </div>
+      <p>
+        {t('block.block')} {blockNumberFormat}
+      </p>
+      <p>{query.data ? parseSimpleDate(query.data?.blockTimestamp) : null}</p>
+    </div>
+  )
+}
+
+export const CellGraph = ({
```

Comment:
How about defining `CellGrpah` and `TxGraph` in two sub modules to make it clear and easy to extend. E.g.

```
/UTXOGraph
  /index  # re-export cell graph and tx graph
  /CellGraph
  /TxGraph
```

Reply:
These two graphs will be switched between them. If export `CellGraph` and `TxGraph` it will expose more detail for the caller.

Reply:
`CellGraph` and `TxGraph` are exported from `index.tsx`, so `export` should be removed  from line 253 and 627 if they are not intended for the public.

And they can be hidden in the `/UTXOGraph` module even they are defined in sub-module of `/UTXOGraph` when they are not re-exported in `/UTXOGraph/index.tsx`

```
/UTXOGraph
  /index.tsx # export a component consisting of Cell Graph and Tx Graph
  /CellGraph.tsx
  /TxGraph.tsx
```